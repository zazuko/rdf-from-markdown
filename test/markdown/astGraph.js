import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import { resolve } from 'path'
import { getHeadersTree } from '../../src/markdown/headersTree.js'
import { createMarkdownParser } from '../../src/markdown/markdownParser.js'
import { getText } from '../support/readFiles.js'

expect.extend({ toMatchSnapshot })

describe('astGraph', async function () {

  it('should produce a graph from an ast', async function () {
    const fullText = await getText(
      { path: resolve('./test/support/roles-mini.md') })
    const parser = createMarkdownParser()
    const ast = await parser.parse(fullText)
    const headersTree = await getHeadersTree({ astNode: ast, fullText })
    expect(headersTree).toMatchSnapshot(this)
  })

})
