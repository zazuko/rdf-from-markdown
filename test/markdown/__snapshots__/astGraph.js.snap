// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`astGraph should produce a graph from an ast 1`] = `
Object {
  "# Roles": Object {
    "## Pancake maker #role": Object {
      "### Purpose": Object {
        "raw": Array [
          "To make the best Pancakes in the world",
        ],
      },
      "### Song": Object {
        "raw": Array [
          "Pancakes, Pancakes !",
        ],
      },
      "### Tasks": Object {
        "raw": Array [
          Array [
            "- Mix ingredients",
            "- Heat the pan",
            "- Prepare the Pancakes",
          ],
          "And a paragraph, explaining even more things",
        ],
      },
      "### Who": Object {
        "raw": Array [
          Array [
            "- Bob",
            "- Alice",
          ],
        ],
      },
      "raw": Array [
        "Some text for the Pancake maker",
      ],
    },
    "## Pancake seller #role": Object {
      "### Purpose": Object {
        "raw": Array [
          "Sell the Pancakes to the customers",
        ],
      },
      "### Tasks": Object {
        "raw": Array [
          Array [
            "- Coordination of Pancake economic activities",
            "- Advertise the Pancakes",
          ],
        ],
      },
      "### Who": Object {
        "raw": Array [
          Array [
            "- Charlie",
          ],
        ],
      },
    },
  },
  "raw": Array [
    "---
description: Simple text
---",
  ],
}
`;
